command/backtrace.sh

```diff
@@ -124,12 +124,17 @@ function _Dbg_do_backtrace {
 
     typeset    filename
     typeset -i adjusted_pos
-    # Position 0 is special in that get the line number not from the
-    # stack but ultimately from LINENO which was saved in the hook call.
     if (( frame_start == 0 )) ; then
 	((count--)) ;
 	adjusted_pos=$(_Dbg_frame_adjusted_pos 0)
-	filename=$(_Dbg_file_canonic "${BASH_SOURCE[$adjusted_pos]}")
+	filename="${BASH_SOURCE[$adjusted_pos]}"
+	resolved_filename="${_Dbg_file2canonic[$filename]}"
+	if [[ ! -z "${resolved_filename}" ]] ; then
+	    filename="$resolved_filename"
+	else
+	    filename=$(_Dbg_file_canonic "$filename")
```

Comment:
Probably needs quotes for whitespace handling.
```suggestion
	    filename="$(_Dbg_file_canonic "$filename")"
```